_posts/2015-09-04-ditching-django-rest-framework-serializers-for-serpy.md

```diff
@@ -0,0 +1,65 @@
+---
+layout: post
+title: Ditching Django REST Framework Serializers for Serpy
+author: Clark
+---
+
+This is the story of how we ditched [Django REST Framework
+Serializers](http://www.django-rest-framework.org/api-guide/serializers/) for
+[serpy](https://github.com/clarkduvall/serpy) and got a performance boost in almost all of our API
+endpoints.
+
+Here at [BetterWorks](https://betterworks.com), we use a lot of tools to keep tabs on our app and
+make sure our API is performing quickly and efficiently. In production, we use [New
+Relic](http://newrelic.com), and locally we use tools like [Django Debug
+Toolbar](https://django-debug-toolbar.readthedocs.org) and Python's
+[profilers](https://docs.python.org/2/library/profile.html). Many times, an endpoint is executing
+too many queries, or the queries are inefficient. These problems are usually pretty easy to solve.
+Other times, our Python code is the problem.
+
+As I was profiling some of our slower APIs, I went through my checklist:
+
+  - Too many SQL queries happening? __NOPE__
+  - Slow SQL queries happening? __NOPE__
+  - A lot of obvious work being done in Python? __NOPE__
+
+A typical view I was seeing this pattern for looked something like this:
```

Comment:
This phrasing is a little hard to parse, esp. for people who don't immediately know you're talking about Django views